tests/test_crawl.py

```diff
@@ -1,8 +1,12 @@
 import json
 import socket
+import logging
+
+from testfixtures import LogCapture
```

Comment:
I like unittest2 too. Problem right now is that some TestCases can't be changed to inherit directly from unittest2.TestCase (or even unittest.TestCase) instead of twisted.trial.unittest.TestCase because the twisted one implements the logic to wait for deferreds, so some tests that use defer.inlineCallbacks are going to return immediately after switching those parent classes (and possibly be marked as successful tests without checking anything).

I just realized that after verifying that the plugin pytest-twisted won't work on test methods from unittest.TestCase objects (that plugin implements the logic to wait for deferreds), since pytest handles separately its integration with unittest. I'd probably need to submit another PR checking that unittest.TestCase is used only when not dealing with deferreds, so I prefer that this PR uses testfixtures package for now.